src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -234,19 +234,22 @@ QgsRasterBlock *QgsSingleBandPseudoColorRenderer::block( int bandNo, QgsRectangl
   }
 
   QRgb myDefaultColor = NODATA_COLOR;
+  QRgb *outputBlockData = outputBlock->colorData();
+  QgsRasterShaderFunction *fcn = const_cast<QgsRasterShaderFunction *>( mShader->rasterShaderFunction() );
```

Comment:
Shouldn't QgsRasterShaderFunction::shade be const to avoid this? From a quick look this should be just marking shade as const, and the mLUT* variables in QgsColorRampShader as mutable (which is valid, because they are cache variables)

Reply:
Yeah good point, I was just being lazy again :-) fixed in b946780